### DISCUSSION TITLE
How to return from OnBeforeLoadAsync when calling AddImageSharp options

### DISCUSSION THREAD

**Discussion by rndthoughts** [SixLabors/ImageSharp.Web#340] [Q&A] [answered]
Can someone give me an example of how to return from OnBeforeLoadAsync:

```
 builder.Services.AddImageSharp(options =>
                {
                    options.OnBeforeLoadAsync = (context, config) =>
                    {
                        context.Commands.Add("width", "240");
                        context.Commands.Add("height", "240");
                    };
                })
```

The above sample code should add the width and height commands to the incoming request pre-processing but I cant figure out what to return.  It seems to suggest i need to return a Task<DecoderOptions?> but I dont know where to get it from or how to construct one?

**Comment by JimBobSquarePants** [comment, marked as answer]
You actually want to use `OnParseCommandsAsync` for this. 

See the example in the docs.

https://docs.sixlabors.com/articles/imagesharp.web/gettingstarted.html
